src/test/java/hudson/plugins/git/browser/casc/GitLabConfiguratorTest.java

```diff
@@ -0,0 +1,136 @@
+        assertEquals(expectedMapping.getScalarValue("version"), described.getScalarValue("version"));
+    }
+
+    @Test
+    public void testInstance() throws Exception {
+        final GitLab expectedConfiguration = new GitLab("http://fake", "2.0");
+        final Mapping mapping = new Mapping();
+        mapping.put("repoUrl", "http://fake");
+        mapping.put("version", "2.0");
+
+        final GitLab instance = configurator.instance(mapping, NULL_CONFIGURATION_CONTEXT);
+        assertEquals(expectedConfiguration.getRepoUrl(), instance.getRepoUrl());
+        assertEquals(String.valueOf(expectedConfiguration.getVersion()), String.valueOf(instance.getVersion()));
+    }
+
+    @Test
+    public void testInstanceWithEmptyRepo() throws Exception {
+        final GitLab expectedConfiguration = new GitLab("", "2.0");
+        final Mapping mapping = new Mapping();
+        mapping.put("repoUrl", "");
+        mapping.put("version", "2.0");
+
+        final GitLab instance = configurator.instance(mapping, NULL_CONFIGURATION_CONTEXT);
+        assertEquals(expectedConfiguration.getRepoUrl(), instance.getRepoUrl());
+        assertEquals(String.valueOf(expectedConfiguration.getVersion()), String.valueOf(instance.getVersion()));
+
+    }
+
+    @Test
+    public void testInstanceWithNullRepo() throws Exception {
+        final GitLab expectedConfiguration = new GitLab(null, "2.0");
+        final Mapping mapping = new Mapping();
+        mapping.put("version", "2.0");
+
+        final GitLab instance = configurator.instance(mapping, NULL_CONFIGURATION_CONTEXT);
+        assertThat(instance.getRepoUrl(), isEmptyString());
+        assertEquals(String.valueOf(expectedConfiguration.getVersion()), String.valueOf(instance.getVersion()));
+    }
+
+
+    @Test
+    public void testInstanceWithEmptyVersion() throws Exception {
+        final GitLab expectedConfiguration = new GitLab("http://fake", "");
+        final Mapping mapping = new Mapping();
+        mapping.put("repoUrl", "http://fake");
+        mapping.put("version", "");
+
+        final GitLab instance = configurator.instance(mapping, NULL_CONFIGURATION_CONTEXT);
+        assertEquals(expectedConfiguration.getRepoUrl(), instance.getRepoUrl());
+        assertEquals(String.valueOf(expectedConfiguration.getVersion()), String.valueOf(instance.getVersion()));
+    }
+
+    @Test
+    public void testInstanceWithNullVersion() throws Exception {
+        // If passing a null, GitLab throws an exception
+        final GitLab expectedConfiguration = new GitLab("http://fake", "");
+        final Mapping mapping = new Mapping();
+        mapping.put("repoUrl", "http://fake");
+
+        final GitLab instance = configurator.instance(mapping, NULL_CONFIGURATION_CONTEXT);
+        assertEquals(expectedConfiguration.getRepoUrl(), instance.getRepoUrl());
+        assertEquals(String.valueOf(expectedConfiguration.getVersion()), String.valueOf(instance.getVersion()));
+    }
+
```

Comment:
Test instance() with a null mapping
```suggestion

    @Test
    public void testInstanceWithNullMapping() throws Exception {
        final Mapping mapping = null;
        final GitLab instance = configurator.instance(mapping, NULL_CONFIGURATION_CONTEXT);
        assertEquals("", instance.getRepoUrl());
        assertNull(instance.getVersion());
    }

```